File: Rosetta.R
rosetta <- read.table("/home/lucas/ISGlobal/Gen_Referencies/Gene_references_rosetta.txt", fill = TRUE, fileEncoding = "UTF-8")

rosetize <- function(x){
  result <- c()
  for (i in 1:length(x)){
     if (x[i] %in% rosetta[,3]){
       result[i] <- rosetta[rosetta[,3] == x[i],1][1]
     } else if (x[i] %in% rosetta[,4]){
       result[i] <- rosetta[rosetta[,4] == x[i],1][1]
     } else if (x[i] %in% rosetta[,5]){
       result[i] <- rosetta[rosetta[,5] == x[i],1][1]
     }
  }
  return(result)
  }


rosetta[rosetta[,3] == "MAL1_28s",1][1]

File: rosetta.py
#!/usr/bin/env python

import sys
from tqdm import tqdm

rosetta = {}
with open("/home/lucas/ISGlobal/Gen_Referencies/Gene_references_rosetta.txt", "r+") as file1:	
	for line in file1:
		rosetta[str(line.split("\t")[0].strip())] = line.split("\t")[1:]
	for key, value in rosetta.iteritems():
		value[-1] = value[-1].strip()

# print len(rosetta)
# print rosetta

genes = {}

def translate(file):
	with open(file) as file:
		for line in file:
			genes[(line.strip())] = []

		for i in tqdm(genes):
			if i in str(rosetta.values()):
				for key, value in rosetta.iteritems():
					if i in value:
							# print i
							# print value
							# print "------------------------------------------------------------------"
							genes[i].append(key)
			else:
				genes[i].append("NA")

	with open("rosseta_IDs3.txt", "a+") as outfile:
		for key, value in genes.iteritems():
			outfile.write(key+"\t")
			first = True
			for i in value:
				if first: 
					outfile.write(i)
					first = False
				else:
					outfile.write(","+i)

			outfile.write("\n")



if __name__ == "__main__":
	file = sys.argv[1]
	translate(file)